mocks/service/launchpad/index.js

```diff
@@ -29,12 +29,18 @@ router.get('/devel/*snaps', (req, res) => {
         {
           resource_type_link: `${base}/#snap`,
           git_repository_url: `${prefix}mock-repo-1`,
-          self_link: `${base}/${owner}/+snap/mock-snap-1`
+          self_link: `${base}/${owner}/+snap/mock-snap-1`,
+          name: 'mock-snap-1'
         },
         {
           resource_type_link: `${base}/#snap`,
           git_repository_url: `${prefix}mock-repo-2`,
-          self_link: `${base}/${owner}/+snap/mock-snap-1`
+          self_link: `${base}/${owner}/+snap/mock-snap-2`,
+          name: 'mock-snap-2'
+        },
+        {
+          resource_type_link: `${base}/#snap`,
+          git_repository_url: `${prefix}mock-repo-3`
```

Comment:
No.  They'll have a `name` field, but that's just a unique URL segment which in this case is generated by BSI, and doesn't relate to the name in `snapcraft.yaml`.

We're explicitly intending to create the snap object in Launchpad before a snap name is necessarily known, so you should expect that repositories in this response may well not have a registered snap name as part of the normal workflow.